File: src/main/java/com/nurkiewicz/tsclass/Compiler.java
package com.nurkiewicz.tsclass;

import com.nurkiewicz.tsclass.parser.Parser;
import com.nurkiewicz.tsclass.parser.ast.ClassDescriptor;

import java.io.IOException;
import java.io.InputStream;

public class Compiler {

    public static void main(String[] args) throws IOException {
        final InputStream programStream = Compiler.class.getResourceAsStream("Greeter.ts");
        final ClassDescriptor cls = new Parser().parse(programStream);
        System.out.println(cls);
    }
}

File: src/main/kotlin/com/nurkiewicz/tsclass/Compiler.kt
package com.nurkiewicz.tsclass

import com.nurkiewicz.tsclass.parser.Parser
import java.io.IOException

object Compiler {

    @Throws(IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val programStream = Compiler::class.java.getResourceAsStream("Greeter.ts")
        val cls = Parser().parse(programStream)
        println(cls)
    }
}
